### ISSUE TITLE
Create Windows installers

### ISSUE THREAD

**Issue by lmoureaux** [longturn/freeciv21#303]
Now that the MSYS build is stable, it would be great to have an installer. The easiest is probably CPack ([doc hint](https://gitlab.kitware.com/cmake/community/-/wikis/doc/cpack/Component-Install-With-CPack)) with the Qt Installer Framework or NSIS (both available in the MSYS repos).

The QIF has a better chance to handle the Qt plugin dir nicely and not need a batch script to set it in the environment.

**Comment by jwrober** [comment]
Core Freeciv uses these steps/settings

1. Welcome Page
2. Ack of the GPL v2 License
3. Components (screenshot)
4. Default language
5. User selection (install for me or everyone)
6. Destination folder
7. Start menu entry
8. (files installed
9. Finish

![image](https://user-images.githubusercontent.com/23037938/108930574-c82c5b00-760b-11eb-9d4f-7b2c7813700f.png)

I would assume we would want something similar. However I don't know if we need to do all the languages right now. Seems to be that we are mostly supporting English.

Do we want to offer the download of additional modpacks during install? SongOfDoom music set and RoundSquare tile set come to mind.

**Comment by lmoureaux** [comment]
IIRC the license stuff can be skipped, there's no legal reason for showing it. I don't mind adding extra modpacks and would even think of including some in the repo. Or maybe we need another repo layout to make them less dependent of Freeciv21 releases.

Is it possible to skip the details with a "default install" button? Use `Program Files (x86)/Freeciv21`, install everything, put client directly in main menu, created desktop shortcut... For the detailed version I would make ruleup, freeciv-manual and ruledit optional. The modpack installer should be mandatory :imp: (if you don't want fcmp or the server, just rebuild locally). 

How large are the translations? Maybe just install them all since they've been downloaded anyway? I'll for sure update the French translation at some point.

**Comment by jwrober** [comment]
Ok that makes sense. I also do like how most installers offer a "typical install" and "custom install". Typical does what you are talking about (e.g. default install).  You answered another question, how did you want to break out the tools. Got it. :)

**Comment by jwrober** [comment]
The Wiki article linked talks about the install() directive.  With the TARGETS there is a sub-option called RUNTIME and another called ARCHIVE.

```cmake
install(TARGETS mylib
  ARCHIVE
  DESTINATION lib)
install(TARGETS mylibapp
  RUNTIME
  DESTINATION bin)
```
In our code all we have is `install(TARGETS RUNTIME [some executable]...)` and are not using `ARCHIVE` for the varying libraries. Is this going to be an issue?

**Comment by lmoureaux** [comment]
The [documentation](https://cmake.org/cmake/help/latest/command/install.html#targets) for the `install` command indicates that `ARCHIVE` is for static libraries and `RUNTIME` for executables. The libraries defined in the build system are linked statically into the executables, so they are not needed at runtime. This is what makes the tools binaries so large compared to a Debian install.

I think all you should need is redefine `CMAKE_INSTALL_BINDIR` and define a variable for `${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}` (used in `data`, it maps to `share/freeciv21` and point to the `data` folder on Windows).

Documentation of the `CMAKE_INSTALL_XXX` variables is [here](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html).

**Comment by jwrober** [comment]
I've got two commits staged in my fork

https://github.com/jwrober/freeciv21/commits/master

I am getting an error for the last command `cpack -G NSIS`

```CPack Error: CPack project name not specified```

For full disclosure, these are the commands in order I am using to build

mkdir build
cd build
cmake -G Ninja ..
ninja
ninja dist-win-exe

The last line does this `add_custom_target(dist-win-exe COMMAND ${CMAKE_CPACK_COMMAND} -G NSIS)`

I am completely stumped and can't get past the error message.

**Comment by lmoureaux** [comment]
Maybe [this](https://cmake.org/cmake/help/latest/module/CPack.html#variables-common-to-all-cpack-generators)? But it should pick up `freeciv21` from the project name.

**Comment by lmoureaux** [comment]
Fixed in https://github.com/lmoureaux/freeciv21/commit/9152760fec5d8fb9d80b945488359f911d72a94d

**Comment by lmoureaux** [comment]
Fixed in #333
